### PR TITLE
Library build task

### PR DESCRIPTION
**Pull request by jmazanec15**
*Issue #, if available:*
#35 

*Description of changes:*
Adding task to build JNI library. Adding it as a dependency of test and integTest tasks so that library is built before tests are run.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -109,13 +109,26 @@ thirdPartyAudit.enabled = false
...
+    workingDir 'jni'
+    commandLine 'cmake', '.'
+}
+
+task buildJniLib(type:Exec) {
+    dependsOn cmakeJniLib
+    workingDir 'jni'
+    commandLine 'make'
+}
+
```
Comment (vamshin) [inline build.gradle:126]: why not club this to one task

task buildJniLib(type:Exec) {
    workingDir 'jni'
    commandLine 'cmake', '.'
     commandLine 'make'
}

**Review by vamshin — COMMENTED** [review]
Also can we add print statement in ```cmake ``` to output the location where we are dumping the shared library?

File: build.gradle (hunk 1 above)
Comment (jmazanec15) [reply build.gradle:126]: The following syntax does not work. `commandLine 'cmake', '.'` is overridden with `commandLine 'make'`. 
```
task buildJniLib(type:Exec) {
workingDir 'jni'
commandLine 'cmake', '.'
commandLine 'make'
}
```

Additionally, having 2 separate tasks allows a user to just run the cmake task if they want.

**Comment by jmazanec15** [comment]
> Also can we add print statement in `cmake ` to output the location where we are dumping the shared library?

Sure.

**Review by vamshin — APPROVED** [review]
LGTM! Building library part of gradlew build makes life easy for developers.

**Pull request merged**